webapp/src/ts/services/enketo.service.ts

```diff
@@ -808,21 +844,71 @@ export class EnketoService {
   }
 
   unload(form) {
-    $(window).off('.enketo-pagemode');
-    if (form) {
-      form.resetView();
+    try {
+      $(window).off('.enketo-pagemode');
+      if (form) {
+        form.resetView();
+      }
+      // unload blobs
+      this.objUrls.forEach((url) => {
+        (window.URL || window.webkitURL).revokeObjectURL(url);
+      });
+
+      delete window.CHTCore.debugFormModel;
+      delete this.currentForm;
+      this.objUrls.length = 0;
+
+      this.unloadDeferred?.resolve();
+    } catch (error) {
+      this.unloadDeferred?.reject(error);
+    }
+  }
+
+  private isTrainingCard(formInternalId) {
+    return formInternalId?.startsWith(TRAINING_PREFIX);
```

Comment:
> I'm using your branch of cht-conf to upload the form you use in the e2e test

I'm going to update the e2e

> but the internal id never gets the training prefix

I think cht-conf doesn't do it at all, I tested by uploading a contact form, and my doc didn't get the prefix `contact:` in the `internalId`. That'd be a good thing to have.

> We filter training forms by their UUID to display them, can we reuse the uuid here, instead of the internal id?

I tried that before, but it needs more refactoring in other places to make it consistent. At this point, I prefer to do those refactorings in a separate ticket for dedicated AT in contacts/reports/training cards.
I've created an issue here: https://github.com/medic/cht-conf/issues/537

Reply:
> As a consequence, this code returns false and the training will show un indefinitely.

Can we filter out forms that have an internalId without the `training:` prefix? So then we make sure we never have a training that keeps on displaying with every reload. We should add some way of reporting to deployment admins that one of their trainings is skipped because of incorrect structure!

Reply:
yes, that's easy!

Reply:
In light of this comment, could you please update the e2e to also test that a completed training is not displayed again?

Reply:
Filter done. Working on the e2e

Reply:
e2e done